### ISSUE TITLE
[BUG] [a11y] SfListItem should have SfButton inside the default slot

### ISSUE THREAD

**Issue by martaradziszewska** [vuestorefront/storefront-ui#1090]
**Describe the bug**
SfListItem should have tabindex because eventually it is a clickable element. In many cases SfListItem have inside SfButton like here:

```
<SfDropdown :is-open="isOpen" :title="title" @click:close="isOpen = !isOpen">
   <SfList>
       <SfListItem v-for="(action, key) in actionList" :key="key">
         <SfButton class="sf-button--full-width sf-button--underlined color-primary" @click="isOpen = 
              !isOpen">{{ action }}
         </SfButton>
       </SfListItem>
    </SfList>
</SfDropdown>

```

As a result we have to double `tab` to move to another element.

**Other examples:**
SfMegaMenu which contains clickable element.
```
<SfListItem v-for="item in items" :key="item">
      <!-- @slot Custom menu-item markup -->
         <slot name="menu-item" v-bind="{ change, item, active }">
            <div @click="change(item)">
              <SfMenuItem :label="item" class="sf-mega-menu__menu-item" />
            </div>
     </slot>
</SfListItem>
```

In. SfContentPages SfListItem puts SfMenuItem in slot which is clickable and should be a button.
```
<SfListItem
              v-for="page in category.items"
              :key="page.title"
              class="sf-content-pages__list-item"
            >
              <!-- @slot Custom menu-item markup -->
              <slot name="menu-item" v-bind="{ updatePage, page, active }">
                <SfMenuItem
                  :class="{ 'sf-menu-item--is-active': page.title === active }"
                  :label="page.title"
                  class="sf-content-pages__menu"
                  @click="updatePage(page.title)"
                >
```
**How to reproduce**

Actually we are covering tabindex issues and v-focus so not every component is updated:
1. Add `tabindex="0"` to `<div>` inside `SfListItem`
2. Go to 'SfDropdown'
3. Start tabbing
4. See error - you have to double tab to move to another element

**Expected behavior**
`SfListItem` should have inside `SfButton` 

**Code examples**
```
<SfListItem>
   <slot>
      <SfButton />
   </slot>
</SfListItem>
```

**Comment by justyna-13** [comment]
There can be cases where SfListItem shouldn't be a button
